### PR TITLE
Fix Supervisor logs fallback

### PR DESCRIPTION
**Pull request by sairon**
## Proposed change

Supervisor logs fallback in get_supervisor_logs didn't work properly because the exception was caught in api_process_raw instead. This was not discovered in tests because the side effect raised OSError, which isn't handled there.

To address that, I split the advanced_logs to two functions, one being a wrapped API handler, one being plain function returning response without any additional error handling. The tests now check for both cases of errors (HassioError and random generic Python error).

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue:
- This PR is related to issue: #5021
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Ruff (`ruff format supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by mdegat01 — CHANGES_REQUESTED** [review]
We should avoid letting unhandled errors like `OSError` bubble up if we don't have to. Rather then changing then changing the API methods like this, couldn't we just add `OSError` to the `except` line here?
https://github.com/home-assistant/supervisor/blob/b09230a884d93eff0f8baf7db60f39d061af2710/supervisor/host/logs.py#L161-L164

Or if you don't think that should raise `HostServiceError` then catch it separately and raise a more appropriate exception instead that extends from `HassioError`. And take whatever other actions you think are required first (log, etc.)

**Comment by sairon** [comment]
> We should avoid letting unhandled errors like `OSError` bubble up if we don't have to.

It wouldn't actually bubble up in the test case - it will be caught by the "too broad" exception cause.

> Rather then changing then changing the API methods like this, couldn't we just add `OSError` to the `except` line here?

The `OSError` is actually one of the random (yet possible) exception classes, so catching that is not the solution.

> Or if you don't think that should raise `HostServiceError` then catch it separately and raise a more appropriate exception instead that extends from `HassioError`. And take whatever other actions you think are required first (log, etc.)

To sum it up - the broad exception handler is there to make sure that `get_supervisor_logs` always returns some output if possible - because without Supervisor logs, you're totally in the dark. This is really just to handle rare edge cases - and if those happen, you should still see the error in the output through the `_LOGGER.exception` called there. We **do not** want this to be handled in any of the upper layers because it is there just as a fallback. If something is really screwed up, you'll get another stack trace within the exception handler.

What I realize now, knowing the Supervisor codebase a bit more, I think that a call to `capture_exception` in the generic `Exception` handler is what is _really_ necessary, so we learn about those those rather rare cases that are caught in Sentry. Am I right?

**Review by mdegat01 — APPROVED** [review]

File: supervisor/api/__init__.py [hunk 1]
```diff
@@ -401,7 +401,7 @@ def _register_supervisor(self) -> None:
 
         async def get_supervisor_logs(*args, **kwargs):
             try:
-                return await self._api_host.advanced_logs(
+                return await self._api_host.advanced_logs_handler(
                     *args, identifier="hassio_supervisor", **kwargs
                 )
             except Exception as err:  # pylint: disable=broad-exception-caught
```
Comment (agners) [inline supervisor/api/__init__.py:407]: Now that we have a known use case (as this PR got triggered by devcontainer usage, see #5021), can we handle this specific and make it not report to Sentry?

I guess the exact error here might depond on weather systemd-journal-gatewayd is available? E.g. would the exception be different if we were to revert https://github.com/home-assistant/devcontainer/pull/6?

Comment (agners) [reply supervisor/api/__init__.py:407]: Let's get this handled in a follow up PR. This PR does what it advertises: Fix the fallback as it is intended currently.

**Review by agners — APPROVED** [review]

**Pull request merged**